core/aws_ddk_core/stages/firehose_s3.py

```diff
@@ -0,0 +1,221 @@
+            Existing Delivery Stream to use in this stage
+        bucket_name: Optional[str] = None
+            Name of S3 Bucket to be created as a delivery destination
+        bucket: Optional[IBucket] = None
+            Existing S3 Bucket to use as a destination for the Firehose Stream.
+            If no bucket is provided, a new one is created
+        buffering_interval: Optional[Duration] = None
+            The length of time that Firehose buffers incoming data before delivering it to the S3 bucket.
+            Minimum: Duration.seconds(60)
+            Maximum: Duration.seconds(900)
+            Default: Duration.seconds(300)
+        buffering_size: Optional[Size] = None
+            The size of the buffer that Kinesis Data Firehose uses for incoming
+            data before delivering it to the S3 bucket.
+            Minimum: Size.mebibytes(1)
+            Maximum: Size.mebibytes(128)
+            Default: Size.mebibytes(5)
+        compression: Optional[Compression] = None
+            The type of compression that Kinesis Data Firehose uses to compress
+            the data that it delivers to the Amazon S3 bucket.
+            Default: - UNCOMPRESSED
+        data_stream: Optional[Stream] = None
+            Existing Kinesis Data Stream to use in stage before Delivery Stream.
+            Setting this parameter will override any creation of Kinesis Data Streams
+            in this stage. `enable_data_stream` will have no effect.
+        enable_data_stream: Optional[bool] = False,
+            Enable Kinesis Data Stream to fron the Firehose delivery stream.
+            Default: false
+        data_output_prefix: Optional[str] = None
+            A prefix that Kinesis Data Firehose evaluates and adds to records before writing them to S3.
+            This prefix appears immediately following the bucket name.
+            Default: “YYYY/MM/DD/HH”
+        enable_data_stream: Optional[bool] = False
+            Add Kinesis Data Stream to front Firehose Delivery.
+            Default: false
+        encryption_key: Optional[IKey] = None
+            The AWS KMS key used to encrypt the data delivered to your Amazon S3 bucket
+        error_output_prefix: Optional[str] = None
+            prefix that Kinesis Data Firehose evaluates and adds to failed records before writing them to S3.
+            This prefix appears immediately following the bucket name.
+            Default: “YYYY/MM/DD/HH”
+        logging: Optional[bool] = True
+            If true, log errors when data transformation or data delivery fails.
+            If `log_group` is provided, this will be implicitly set to true.
+            Default: true - errors are logged.
+        log_group: Optional[ILogGroup] = None
+            The CloudWatch log group where log streams will be created to hold error logs.
+            Default: - if logging is set to true, a log group will be created for you.
+        alarm_threshold: Optional[int] = 900
```

Comment:
Should we be specific in the alarm naming (i.e. kinesis_delivery_stream) to keep the door open for potential alarms in the future?

Reply:
Definitely.